src/ServicePulse.Host/app/modules/monitoring/js/directives/ui.particular.monitoringConnectivityStatus.js

```diff
@@ -0,0 +1,58 @@
+﻿
+; (function (window, angular, $, undefined) {
+    'use strict';
+
+
+    function controller($scope, connectivityNotifier, monitoringService, $interval) {
+        $scope.isSCMonitoringConnecting = true;
+        connectivityNotifier.getConnectionStatusSource().subscribe(value => {
+            $scope.isSCMonitoringConnected = value;
+            $scope.isSCMonitoringConnecting = false;
+        });
+
+        var scMonitoringConnectionPing = $interval(function () {
+            var promises = monitoringService.checkConnections();
```

Comment:
```suggestion
            var promises = monitoringService.checkConnections().map(p => {
                p.then(r => {
                    connectivityNotifier.reportSuccessfulConnection(i);
                }, e => {
                    connectivityNotifier.reportFailedConnection(i);
                });
            });
            
```

Instead of manually iterating over the array